### ISSUE TITLE
BAMultiShapesGenerator

### ISSUE THREAD

**Issue by steveazzolin** [pyg-team/pytorch_geometric#6545]
### 🚀 The feature, motivation and pitch

As suggested by @rusty1s in #6541 I'll include the `BAMultiShapesGenerator` in a separate PR.

However, @rusty1s suggested that adding it as a Graph `Generator` might not be the best solution.
I think that a natural way to handle it would be via something like `ExplainerDataset`.
However, as already said before, the current implementation does not seem very suitable for a nice incorporation of that.
Maybe changing `ExplainerDataset` into `NodeClassificationExplainerDataset` (keeping the current implementation) and adding a new `GraphClassificationExplainerDataset` would be nicer. 

Which are your opinions?

### Additional context

This is a followup to @rusty1s's comment in #6541

**Comment by rusty1s** [comment]
Yes, this is exactly what I was referring to. Ultimately, this should be a `Dataset` rather than a `GraphGenerator`. Something like `GraphLevelExplainerDataset` sounds good to me.

**Comment by steveazzolin** [comment]
Great! I'll proceed in this way then.
Thanks
